core/src/executive/internal_contract/contracts/admin.rs

```diff
@@ -15,21 +15,23 @@ use crate::{
     evm::{ActionParams, Spec},
     impl_function_type, make_function_table, make_solidity_contract,
     make_solidity_function, rename_interface,
-    state::{State, Substate},
+    state::{StateGeneric, Substate},
     vm,
 };
+#[cfg(test)]
+use cfx_storage::StorageState;
+use cfx_storage::StorageStateTrait;
 use cfx_types::{Address, U256};
 #[cfg(test)]
 use rustc_hex::FromHex;
 
-lazy_static! {
-    static ref CONTRACT_TABLE: SolFnTable =
-        make_function_table!(SetAdminSnake, Destroy);
-    static ref CONTRACT_TABLE_V2: SolFnTable =
-        make_function_table!(SetAdmin, Destroy, GetAdmin);
+fn generate_fn_table<S: StorageStateTrait + Send + Sync + 'static>(
+) -> SolFnTable<S> {
+    make_function_table!(SetAdmin<S>, Destroy<S>, GetAdmin<S>)
```

Comment:
I wanted to make this into something like this:

```rust
fn generate_fn_table<S: StorageStateTrait + Send + Sync + 'static>() -> &SolFnTable<S> {
    static t: SolFnTable<S> = make_function_table!(SetAdmin<S>, Destroy<S>, GetAdmin<S>);
    &t
}
```

Unfortunately, Rust does not allow this (`can't use type parameters from outer function`) and I could not find a way to do *generic statics* on stable.